src/test/scala/linting/rule/RenameDesiredNamesSpec.scala

```diff
@@ -0,0 +1,189 @@
+      DesiredNameAnnotation("BarWith1Input", top.module("BarWith1Input")),
+      OverrideDesiredNameAnnotation("BarWith2Inputs", top.module("Bar_2")),
+      OverrideDesiredNameAnnotation("BarWith2Inputs", top.module("Bar_3"))
+    ))
+  }
+
+  property("It should keep modules names stable between runs") {
+    val top = CircuitTarget("Foo")
+    test(
+      TestCase(
+        """|circuit Foo:
+           |  module Bar_1:
+           |    output in1: UInt<1>
+           |  module Bar_2:
+           |    output in1: UInt<1>
+           |    output in2: UInt<1>
+           |  module Bar_3:
+           |    output in1: UInt<1>
+           |    output in2: UInt<1>
+           |    output in3: UInt<1>
+           |  module Foo:
+           |    inst bar_1 of Bar_1
+           |    inst bar_2 of Bar_2
+           |    inst bar_3 of Bar_3
+           |""".stripMargin,
+        Seq(
+          UnstableNameAnnotation(top.module("Bar_1")),
+          StableNameAnnotation(top.module("Bar_2")),
+          StableNameAnnotation(top.module("Bar_3")),
+          OverrideDesiredNameAnnotation("Bar_2", top.module("Bar_2")),
+          OverrideDesiredNameAnnotation("Bar_3", top.module("Bar_3"))
+        )
+      ),
+      TestCase(
+        """|circuit Foo:
+           |  module Bar_1:
+           |    output in1: UInt<2>
+           |  module Bar_5:
+           |    output in1: UInt<1>
+           |    output in2: UInt<1>
+           |  module Bar_6:
+           |    output in1: UInt<1>
+           |    output in2: UInt<1>
+           |    output in3: UInt<1>
+           |  module Foo:
+           |    inst bar_1 of Bar_1
+           |    inst bar_5 of Bar_5
+           |    inst bar_6 of Bar_6
+           |""".stripMargin,
+        Seq(
+          UnstableNameAnnotation(top.module("Bar_1")),
+          StableNameAnnotation(top.module("Bar_5")),
+          StableNameAnnotation(top.module("Bar_6")),
+          OverrideDesiredNameAnnotation("Bar_4", top.module("Bar_1")),
+          OverrideDesiredNameAnnotation("Bar_2", top.module("Bar_5")),
+          OverrideDesiredNameAnnotation("Bar_3", top.module("Bar_6"))
+        )
+      )
+    )
+  }
+
+  property("It should error if renaming to an already existing module name") {
+    val top = CircuitTarget("Foo")
+    val testCase = TestCase(
+      """|circuit Foo:
+         |  module Bar_1:
+         |    output in1: UInt<1>
+         |  module Bar_2:
+         |    output in1: UInt<1>
+         |    output in2: UInt<1>
+         |  module Bar_3:
+         |    output in1: UInt<1>
+         |    output in2: UInt<1>
+         |    output in3: UInt<1>
+         |  module Foo:
+         |    inst bar_1 of Bar_1
+         |    inst bar_2 of Bar_2
+         |    inst bar_3 of Bar_3
+         |""".stripMargin,
+      Seq(OverrideDesiredNameAnnotation("Bar_1", top.module("Bar_2")))
+    )
+    an [Exception] should be thrownBy renameNames(testCase)
+  }
+}
```

Comment:
It would be nice to see a pair of tests - one where we tried to use OverrideDesiredName annotation then run the linter, but failed (because our desired names were the same for two different modules), then how we fixed it by using a more sophisticated OverrideDesiredName. For example maybe we had `BarWithTwoInputs` that needed. to be `BarWithTwoBoolInputs` and `BarWithTwoUIntInputs`.

Reply:
added to `LintConflictingModuleNamesSpec`

Reply:
Sorry, did you push that change? I did not see it

Reply:
https://github.com/chipsalliance/rocket-chip/pull/2452/files#diff-f5acd5259051dcdd9ad22c3a5b5a6ba6R58-R111